CDTDatastore/touchdb/TDReplicator.m

```diff
@@ -800,10 +863,35 @@ - (void)saveLastSequence
     _lastSequenceChanged = _overdueForSave = NO;
 
     CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@ checkpointing sequence=%@", self, _lastSequence);
-    NSMutableDictionary* body = [_remoteCheckpoint mutableCopy];
-    if (!body) body = $mdict();
-    [body setValue:_lastSequence forKey:@"lastSequence"];
-    
+    NSMutableDictionary* body = [self.remoteCheckpoint mutableCopy];
+
+    // upgrade the replication doc to the new format.
+    if (body && !body[@"history"]) {
```

Comment:
It can do, but I think the code will be same, since we don't need to check the number since we haven't saved one in the docs before.